src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -0,0 +1,165 @@
+import { AfterViewInit, Component, ElementRef, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { ItemData } from '../../services/item-datasource.service';
+import { CompleteFunction, ErrorFunction, Platform, Task, TaskParams, TaskProxyManager } from 'src/app/shared/task/task-xd-pr';
+import { interval, Subscription } from 'rxjs';
+
+interface TaskTab {
+  name: string
+}
+
+@Component({
+  selector: 'alg-item-display',
+  templateUrl: './item-display.component.html',
+  styleUrls: [ './item-display.component.scss' ]
+})
+export class ItemDisplayComponent implements OnInit, AfterViewInit, OnDestroy {
+  @Input() itemData?: ItemData;
+  @ViewChild('iframe') iframe?: ElementRef<HTMLIFrameElement>;
+
+  state : 'loading' | 'loaded' | 'unloading';
+  url? : SafeResourceUrl;
+  msg = '';
+
+  tabs: TaskTab[] = [];
+  activeTab: TaskTab;
+
+  taskProxyManager: TaskProxyManager;
+  task?: Task;
+  platform? : Platform;
+
+  height: number;
+  heightInterval? : Subscription;
+
+  lastAnswer = '';
+  lastState = '';
+  saveInterval? : Subscription;
+
+  constructor(private sanitizer: DomSanitizer) {
+    this.state = 'loading';
+    this.height = 400;
+    const initialTab = { name: 'Task' };
+    this.tabs = [ initialTab ];
+    this.tabs.push({ name: 'Editor' });
+    this.activeTab = initialTab;
+    this.taskProxyManager = new TaskProxyManager();
+  }
+
+
+  // Lifecycle functions
+  ngOnInit(): void {
+    const url = this.itemData?.item.url || '';
+    // TODO get sToken
+    const sToken = '';
+    this.setUrl(this.taskProxyManager.getUrl(url, sToken, 'http://algorea.pem.dev', 'task-'));
+  }
+
+  ngAfterViewInit(): void {
+    if (this.iframe) {
```

Comment:
`this.iframe` can be unset in the first call to ngAfterViewInit, because the ViewChild doesn't set `this.iframe` straight away. We just wait until `this.iframe` is set and then load, as another call to ngAfterViewInit is done after the ViewChild is actually initialized.